### PR TITLE
Improve derivation of ToCss again

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17662)

### CODE REVIEW

**Comment by nox** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=7260af46027850b45445223487955dc054eee9f9

File: components/style/properties/longhand/box.mako.rs [hunk 1]
```diff
@@ -900,31 +896,35 @@ ${helpers.predefined_type(
...
-                Skew(ax, Some(ay)) => write!(dest, "skew({}, {})", Css(ax), Css(ay)),
-                SkewX(angle) => write!(dest, "skewX({})", Css(angle)),
-                SkewY(angle) => write!(dest, "skewY({})", Css(angle)),
-                Translate(ref tx, None) => write!(dest, "translate({})", Css(tx)),
-                Translate(ref tx, Some(ref ty)) => write!(dest, "translate({}, {})", Css(tx), Css(ty)),
-                TranslateX(ref tx) => write!(dest, "translateX({})", Css(tx)),
-                TranslateY(ref ty) => write!(dest, "translateY({})", Css(ty)),
-                TranslateZ(ref tz) => write!(dest, "translateZ({})", Css(tz)),
-                Translate3D(ref tx, ref ty, ref tz) => write!(
+                SpecifiedOperation::Skew(ax, None) => write!(dest, "skew({})", Css(ax)),
```
Comment (emilio) [inline components/style/properties/longhand/box.mako.rs]: Hmm... it'd e so nice to derive these :-)

File: components/style/properties/longhand/counters.mako.rs [hunk 2]
```diff
@@ -40,14 +39,16 @@
...
 
-        #[derive(Debug, PartialEq, Eq, Clone)]
         #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        #[derive(Clone, Debug, Eq, PartialEq, ToCss)]
         pub enum ContentItem {
             /// Literal string content.
             String(String),
             /// `counter(name, style)`.
+            #[css(comma, function)]
             Counter(String, CounterStyleType),
```
Comment (emilio) [inline components/style/properties/longhand/counters.mako.rs]: This one should be `CustomIdent` afaict.

File: components/style/properties/longhand/box.mako.rs (hunk 1 above)
Comment (nox) [inline components/style/properties/longhand/box.mako.rs]: That's my ultimate plan.

File: components/style/properties/longhand/counters.mako.rs [hunk 3]
```diff
@@ -40,14 +39,16 @@
...
             /// `counters(name, separator, style)`.
+            #[css(comma, function)]
             Counters(String, String, CounterStyleType),
```
Comment (emilio) [inline components/style/properties/longhand/counters.mako.rs]: And the left one here too.

**Review by emilio — APPROVED** [review]
r=me with those fixed

**Comment by nox** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=ea3f1c90e7afb195617ae81390631d101a24f806

**Comment by SimonSapin** [comment]
r=me on new commits since Emilio’s review

---

Reviewed 2 of 2 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 4 of 4 files at r6, 3 of 4 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 6 of 6 files at r10, 1 of 1 files at r11, 3 of 3 files at r12, 1 of 1 files at r13.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/17662#-:-KoqLjt6kkg_Gm0nYDHq:b-9jspie)*

**Comment by nox** [comment]
I forgot `#[css(function)]` on two variants of `TrackSize<T>`.

https://treeherder.mozilla.org/#/jobs?repo=try&revision=9af9242e6e80518dbd79b2c3e48bff5728d7bb96

**Pull request merged**